cherab/tools/plasmas/slab.pyx

```diff
@@ -194,3 +194,67 @@ def build_slab_plasma(length=5, width=1, height=1, peak_density=1e19, peak_tempe
     plasma.composition = species
 
     return plasma
+
+
+def build_constant_slab_plasma(length=5, width=1, height=1, electron_density=1e19, electron_temperature=2.5e3,
+                               ions=[(hydrogen, 1, 1e19, 2.5e3, Vector3D(0, 0, 0))],
+                               b_field=Vector3D(0, 0, 0), parent=None):
+    """
+    Constructs a simple slab of plasma with constant conditions.
+
+    The plasma is defined for positive x starting at x = 0, symmetric in y-z. The plasma
+    parameters such as electron density and temperature are constant over the plasma volume.
+
+    Raysect cannot handle infinite geometry, so overall spatial dimensions of the slab need
+    to be set, [length, width, height]. These can be set very large to make an effectively
+    infinite slab of plasma, although the numerical performance will degrade accordingly.
+    The dimensions should be set appropriately with valid assumptions for your scenario.
+
+    Ion species can be included as a list of tuples, where each tuple specifies an
+    impurity. The specification format is (species, charge, density, temperature, velocity). For example:
+
+        >>> ions = [(carbon, 6, 1e18, 3.4e3, Vector3D(1.0e3, 0, 0))]
+
+    :param float length: the overall length of the slab along x.
+    :param float width: the y width of the slab.
+    :param float height: the z height of the slab.
+    :param float electron_density: the electron density in m^-3 .
+    :param float electron_temperature: the electron temperature in eV.
+    :param list ions: an optional list of impurities to include.
+    :param Vector3D b_field: vector giving the magnetic field
+    :param parent: the Raysect scene-graph parent node.
+
+    .. code-block:: pycon
+
+       >>> from raysect.optical import World
+       >>> from cherab.core.atomic import hydrogen, carbon
+       >>> from cherab.tools.plasmas.slab import build_constant_slab_plasma
+       >>>
+       >>> ions = [(hydrogen, 0, 1e19, 3.5e3, Vector3D(5e3, 0, 0)), (carbon, 5, 1e18, 3.4e3, Vector3D(1.0e3, 0, 0))]
+       >>> plasma = build_constant_slab_plasma(0.2, 0.5, 0.5, electron_density = 1.19, electron_temperature=4e4, ions=ions, b_field=Vector3D(0, 5, 0))
+       >>> plasma.parent = World()
+    """
+
+    zero_velocity = ConstantVector3D(Vector3D(0, 0, 0))
+
+    # create electron distribution
+    e_density = Constant3D(electron_density)
```

Comment:
The `Maxwellian` class will automatically wrap `electron_density` into a `Constant3D` object, so doing it here adds unnecessary boilerplate. This wasn't the case when `build_slab_plasma` was written, but it is now and we should take advantage of this in new code where possible. The same applies to the temperature on the next line.

Unfortunately we don't yet wrap constant vectors in `ConstantVector3D` objects automatically. I've just raised #213 to address this, so perhaps wait until that issue is resolved before making any changes here.

Reply:
I will change this and lets wait with merge until the framework also wraps constant vectors.

Reply:
@Mateasek now that #214 has been merged into development this can be updated. You'll now be able to remove the `Constant3D` and `ConstantVector3D` wrappers here: these constants will be automatically wrapped into function objects. Can you please update this PR to the latest development branch and make these changes? Then it should be good to merge.